File: python/NeetCode150/3_two_int_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}
        i = 0
        for num in nums:
            if num in numMap.keys():
                return [numMap[num], i]
            numMap.update({target-num:i})
            i+=1
        return None

File: java/NeetCode150/3_two_int_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> target_index_map = new HashMap<Integer, Integer>();
        for(int i=0; i<nums.length; i++)
        {
            int index = target_index_map.getOrDefault(nums[i], -1);
            if(index!=-1)
            {
                int[] result = {index, i};
                return result;
            }
            target_index_map.put(target-nums[i], i);
        }
        return null;
    }
}